### PR TITLE
mitie 0.4

### PR DESCRIPTION
**Pull request by dunn**

### CODE REVIEW

File: Library/Formula/mitie.rb [hunk 1]
```diff
@@ -14,27 +14,34 @@ class Mitie < Formula
 
   depends_on :python if MacOS.version <= :snow_leopard
 
+  option "with-models", "Install v0.2 models (~415MB)"
```
Comment (tdsmith) [inline Library/Formula/mitie.rb]: What are these? (Is this useful without the models?)

File: Library/Formula/mitie.rb [hunk 2]
```diff
@@ -14,27 +14,34 @@ class Mitie < Formula
...
 
   def install
-    (share+"MITIE-models").install resource("models-english")
+    if build.with? "models"
+      (share/"MITIE-models").install resource("models-english")
+    end
+
     inreplace "mitielib/makefile" do |s|
-      s.gsub! /libmitie.so/, "libmitie.dylib"
+      s.gsub!(/libmitie.so/, "libmitie.dylib")
```
Comment (tdsmith) [inline Library/Formula/mitie.rb]: fwiw this is fine without parentheses

File: Library/Formula/mitie.rb [hunk 3]
```diff
@@ -14,27 +14,34 @@ class Mitie < Formula
...
+    (lib/"python2.7/site-packages").install "mitielib/mitie.py"
+    (share/"mitie").install "examples", "sample_text.txt",
       "sample_text.reference-output", "sample_text.reference-output-relations"
     bin.install "ner_example", "ner_stream", "relation_extraction_example"
   end
 
   test do
-    system "#{bin}/ner_example", "#{share}/MITIE-models/english/ner_model.dat", "#{share}/mitie/sample_text.txt"
+    system ENV.cc, "-I#{include}", "-L#{lib}", "-lmitie",
+           share/"mitie/examples/C/ner/ner_example.c", "-o", "test"
```
Comment (tdsmith) [inline Library/Formula/mitie.rb]: I guess the models need to be installed to invoke this?

File: Library/Formula/mitie.rb (hunk 1 above)
Comment (dunn) [inline Library/Formula/mitie.rb]: I'm not 100 percent sure! But I assumed something of that size shouldn't be downloaded by default.

File: Library/Formula/mitie.rb (hunk 2 above)
Comment (dunn) [inline Library/Formula/mitie.rb]: Oh, emacs flagged it as "ambiguous first argument". ¯\\\_(ツ)_/¯ Should I revert it?

File: Library/Formula/mitie.rb (hunk 3 above)
Comment (dunn) [inline Library/Formula/mitie.rb]: Yeah, the old test required the models.

File: Library/Formula/mitie.rb (hunk 1 above)
Comment (dunn) [inline Library/Formula/mitie.rb]: Their readme just says [you need the models in order to run the provided examples](https://github.com/mit-nlp/MITIE#initial-setup).

Comment (dunn) [inline Library/Formula/mitie.rb]: And it installs a C library too, so there are definitely use-cases that don't require the models.

Comment (tdsmith) [inline Library/Formula/mitie.rb]: I think the models are useful: they let you do things like http://blog.dlib.net/2014/07/mitie-v02-released-now-includes-python.html?m=1 out of the box. I think making them optional (`option "without-models"`) is fair though.

File: Library/Formula/mitie.rb (hunk 2 above)
Comment (tdsmith) [inline Library/Formula/mitie.rb]: eh, nah, maybe emacs knows better than I do

**Comment by dunn** [comment]
OK, made the models recommended and combined the old and new tests.

File: Library/Formula/mitie.rb [hunk 4]
```diff
@@ -14,27 +14,37 @@ class Mitie < Formula
...
     bin.install "ner_example", "ner_stream", "relation_extraction_example"
   end
 
   test do
-    system "#{bin}/ner_example", "#{share}/MITIE-models/english/ner_model.dat", "#{share}/mitie/sample_text.txt"
+    system ENV.cc, "-I#{include}", "-L#{lib}", "-lmitie",
+           share/"mitie/examples/C/ner/ner_example.c",
+           "-o", testpath/"ner_example"
+    system "./ner_example", share/"MITIE-models/english/ner_model.dat",
+           share/"mitie/sample_text.txt"
```
Comment (xu-cheng) [inline Library/Formula/mitie.rb]: Maybe need to check `built.with? "models"` then run the test.

Comment (dunn) [inline Library/Formula/mitie.rb]: I can do that, but I've been told previously not to worry about the test passing with non-default options.

**Comment by tdsmith** [comment]
:metal:

**Pull request closed without merging**